lib/auth/acl.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+/**
+ * @fileoverview Access Control Lists
+ */
+// config is used below, eslint
+const config = require('config'); // eslint-disable-line no-unused-vars
+const NodeAcl = require('acl');
+
+const userRoles = require('../../config/all/roles');
+
+// memoryBackend starts with a lowercase from node-acl
+const acl = new NodeAcl(new NodeAcl.memoryBackend()); // eslint-disable-line new-cap
+
+/*
+ * Get user roles from configuration
+ *
+ * @param {array} cfg user configuration
+ *
+ * @returns {array} user configuration
+ */
+const roles = (cfg) => {
+  let rolesConfig;
+  let aclRoles;
+  if (cfg) {
+    if (!Array.isArray(cfg)) {
+      throw new Error(config.roles.config.error);
+    }
+    rolesConfig = cfg;
+  }
+  else {
+    rolesConfig = userRoles;
+  }
+
+  if (rolesConfig && Array.isArray(rolesConfig)) {
+    aclRoles = rolesConfig.map((role) => {
+      return {
+        roles: role.id,
+        allows: role.allows,
+      };
+    });
+  }
+
+  return aclRoles;
```

Comment:
This will return `undefined` if `rolesConfig` is not an array or is undefined. Do we want to be returning `undefined`? if we want to return a null value, it should be `null`, otherwise it should be an empty array.